src/NuGet.Core/NuGet.Common/PathUtil/NuGetEnvironment.cs

```diff
@@ -18,7 +18,7 @@ public static class NuGetEnvironment
 #endif
 
         private static readonly Lazy<string> _getHome = new Lazy<string>(() => GetHome());
-
+        private static bool? EnhancedHttpRetryIsEnabled = null;
```

Comment:
I'm starting to think this should just be a static singleton on `HttpRetryHandler` instead of here in `NuGetEnvironment`.  We can move it later if we want to make the feature more broad but for now its only really needed in `NuGet.Protocol` and could just be on `HttpRetryHandler`.  What do you think?

Reply:
I put this here originally at your suggestion, that's fine by me; if it removes it from the public API txt files even better, not sure it does.

Reply:
Yeah sorry I think I meant to have it be an environment variable but didn't mean to put it in this class.  Either way I think it would be better in NuGet.Protocol so we can make it `internal`.